pymc/model_graph.py

```diff
@@ -198,18 +206,32 @@ def get_plates(self, var_names: Optional[Iterable[VarName]] = None) -> Dict[str,
         """
         plates = defaultdict(set)
 
+        # TODO: Evaluate all RV shapes and dim_length at once.
+        #       This should help to find discrepancies, and
+        #       avoids unncessary function compiles for deetermining labels.
+
```

Comment:
This is not needed for the bugfix, but I commented it here for the next person touching this code.